File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.java
class Solution {
    public int tribonacci(int n) {
        if(n<=1)
            return n;
        if(n==2)
            return 1;
        int prev1 =0;
        int prev2 =1;
        int prev3 =1;
        for(int i=3;i<=n;i++){
            int temp = prev3;
            prev3 = temp+prev2+prev1;
            prev1= prev2;
            prev2=temp;
        }
        return prev3;
    }
}

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.py
class Solution:
    def tribonacci(self, n):

        t = [0] * (n + 3)
        t[0], t[1], t[2] = 0, 1, 1

        if n < 2:
            return t[n]

        for i in range(3, n + 1):
            t[i] = t[i - 1] + t[i - 2] + t[i - 3]

        return t[n]
